src/users/userAuthController.ts

```diff
@@ -51,9 +51,15 @@ export const login = catchFunction(
 
 export const getUser = catchFunction(
   async (req: Request, res: Response, next: NextFunction) => {
-    const userId = (req as any).user.userId;
+    const userId = req.user?.userId;
 
-    const user = await getUserById(parseInt(userId));
+    if (typeof userId !== "number") {
+      const error = new AppError("Unauthorized", 401);
+      return next(error);
+    }
```

Comment:
The check here should just be if the user exists or not. 
```suggestion
    if (!userId) {
      const error = new AppError("Unauthorized", 401);
      return next(error);
    }
```